tools/clang/tools/dxcompiler/dxcpdbutils.cpp

```diff
@@ -65,21 +70,139 @@ static HRESULT CopyWstringToBSTR(const std::wstring &str, BSTR *pResult) {
   return S_OK;
 }
 
-static bool ShouldIncludeInFlags(StringRef strRef, bool *skip_next_arg) {
+static std::wstring NormalizePath(const WCHAR *path) {
+  std::string FilenameStr8 = Unicode::UTF16ToUTF8StringOrThrow(path);
+  llvm::SmallString<128> NormalizedPath;
+  llvm::sys::path::native(FilenameStr8, NormalizedPath);
+  std::wstring FilenameStr16 = Unicode::UTF8ToUTF16StringOrThrow(NormalizedPath.c_str());
+  return FilenameStr16;
+}
+
+static bool IsBitcode(const void *ptr, size_t size) {
+  const uint8_t pattern[] = {'B','C',};
+  if (size < _countof(pattern))
+    return false;
+  return !memcmp(ptr, pattern, _countof(pattern));
+}
+
+static bool ShouldIncludeInFlags(const std::wstring &str, bool *skip_next_arg) {
   *skip_next_arg = false;
-  const char *specialCases[] = { "/T", "-T", "-D", "/D", "-E", "/E", };
+  const wchar_t *specialCases[] = { L"/T", L"-T", L"-D", L"/D", L"-E", L"/E", };
   for (unsigned i = 0; i < _countof(specialCases); i++) {
-    if (strRef == specialCases[i]) {
+    if (str == specialCases[i]) {
       *skip_next_arg = true;
       return false;
     }
-    else if (strRef.startswith(specialCases[i])) {
+    else if (str.find(specialCases[i]) == 0) {
```

Comment:
Is this to handle /DDefineWithoutASpaceAfterSlashD ?

Reply:
Yeah, since -T -D -E can all do that.